website/content/en/preview/faq.md

```diff
@@ -95,6 +95,12 @@ If the instance type is unavailable for some reason, then fleet will move on to
 If you are using the spot capacity type, Karpenter uses the capacity-optimized-prioritized allocation strategy which tells fleet to find the instance type that EC2 has the most capacity of which will decrease the probability of a spot interruption happening in the near term.
 See [Choose the appropriate allocation strategy](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-fleet-allocation-strategy.html#ec2-fleet-allocation-use-cases) for information on fleet optimization.
 
+### What if there is no spot capacity? Will Karpenter fallback to on-demand?
+
+Karpenter will fallback to on-demand, if your provisioner specifies both spot and on-demand. 
+
+More specifically, Karpenter maintains a concept of "offerings" which is a combination of instance type, zone, and purchase type. Spot offerings are prioritized, if they're available. Whenever the fleet API returns an insufficient capacity for spot, that offering is removed from the list for 5 minutes. Since fleet gives expressive responses about which offerings are unavilable, it results in at worst 2 API calls. The retry will happen immediately within milliseconds.
```

Comment:
```suggestion
More specifically, Karpenter maintains a concept of "offerings" for each instance type, which is a combination of zone and capacity type (equivalent in the AWS cloud provider to an EC2 purchase option). Spot offerings are prioritized, if they're available. Whenever the Fleet API returns an insufficient capacity error for Spot instances, those particular offerings are temporarily removed from consideration (across the entire provisioner) for 45 seconds so that Karpenter can make forward progress through fallback. The retry will happen immediately within milliseconds.
```

> Since Fleet gives expressive responses about which offerings are unavailable, it results in at worst 2 API calls for that initial failure. 

I think this is confusing language because I'm not sure that it's true. First of all, you're talking about the Fleet API but there are other EC2 APIs called by Fleet. More importantly, I don't really know what this means because if we retry and still get ICEd on OD we'll re-compute the bin packing and try again. Hence, it could be several Fleet API calls. Karpenter does an admirable job though of trying to keep the the number of round trips to Fleet very small though and should often succeed in this effort.

Reply:
I'd remove the specific time we cache for since that's an implementation detail that may be changed I'd also remove the last sentence on the retry happening immediately within milliseconds. 

Since pods are separated by scheduling constraints, if the first group of scheduling fails due to Insufficient Capacity, then the next group will still attempt to provision nodes and will likely take longer than a few milliseconds. Even if there is no other scheduling group, since pods are batched at the start of the provisioning loop, the next run will take at least 1 second since that is the minimum pod batching window duration.